android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/EditTextFactory.java

```diff
@@ -212,6 +218,15 @@ private void attachInfoIcon(String stepName, JSONObject jsonObject, RelativeLayo
 
     }
 
+    private void addEqualsValidator(JsonFormFragment formFragment,JSONObject jsonObject, MaterialEditText editText) throws JSONException {
+        JSONObject requiredObject = jsonObject.optJSONObject(JsonFormConstants.V_EQUALS);
+        if (requiredObject != null) {
+            String referencedValue = requiredObject.optString(JsonFormConstants.VALUE,"");
+            MaterialEditText referencedEditText = (MaterialEditText) formFragment.getJsonApi().getFormDataView(referencedValue);
+            editText.addValidator(new ReferenceFieldValidator(requiredObject.getString(JsonFormConstants.ERR),referencedEditText));
+            FormUtils.setRequiredOnHint(editText);
+        }
+    }
```

Comment:
Changes from this PR https://github.com/opensrp/opensrp-client-native-form/pull/663. I see some requested changes that have not been addressed here https://github.com/opensrp/opensrp-client-native-form/pull/663